src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -219,7 +220,13 @@ export class EvaluationPlugin extends UIPlugin {
     }
     const array = compiledFormula.execute(ranges, ...this.compilationParams);
     if (isMatrix(array)) {
-      return array.map((col) => col.map((row) => row.value));
+      throw new Error("results array are not supported outside the evaluation grid");
+    }
+    if (array.value instanceof EvaluationError) {
+      return array.value.errorType;
+    }
```

Comment:
Seems strange to only here treat error as strings, and not at any other part of the evaluation process.

Also if we want to do that we should test instanceof `Error` (or improve the wrapping of errors into EvaluationError) , because for example WrongNumberOfArgs will return an Error and not an EvaluationError